### ISSUE TITLE
Write a lot of small files in s3 when there are two versions schema data in topic

### ISSUE THREAD

**Issue by harveyyue** [confluentinc/kafka-connect-storage-cloud#520]
Currently, we capture 100 shard tables to a topic in debezium. DBA add four columns to these shard tables one by one, this means there are two version schema in this topic. Unfortunately, dba met the lock table issue, they paused the rest of tables schema change action, we had to write a lot of small files in s3 folder and got the huge lag in kafka consumer processing.

The s3 small files:
2022-06-03 03:26:17   46.4 KiB db_name.db_user_data.tb_all_shards+0+8138464479.snappy.parquet
2022-06-03 03:26:17   31.9 KiB db_name.db_user_data.tb_all_shards+0+8138464550.snappy.parquet
2022-06-03 03:26:17   36.7 KiB db_name.db_user_data.tb_all_shards+0+8138464551.snappy.parquet
2022-06-03 03:26:17   33.1 KiB db_name.db_user_data.tb_all_shards+0+8138464573.snappy.parquet
2022-06-03 03:26:18   54.7 KiB db_name.db_user_data.tb_all_shards+0+8138464575.snappy.parquet
2022-06-03 03:26:18   30.1 KiB db_name.db_user_data.tb_all_shards+0+8138464670.snappy.parquet
2022-06-03 03:26:18   38.3 KiB db_name.db_user_data.tb_all_shards+0+8138464672.snappy.parquet
2022-06-03 03:26:18   39.9 KiB db_name.db_user_data.tb_all_shards+0+8138464693.snappy.parquet
2022-06-03 03:26:19   30.1 KiB db_name.db_user_data.tb_all_shards+0+8138464733.snappy.parquet
2022-06-03 03:26:19   34.8 KiB db_name.db_user_data.tb_all_shards+0+8138464734.snappy.parquet
2022-06-03 03:26:19   40.5 KiB db_name.db_user_data.tb_all_shards+0+8138464736.snappy.parquet
2022-06-03 03:26:19   34.8 KiB db_name.db_user_data.tb_all_shards+0+8138464770.snappy.parquet
2022-06-03 03:26:20   51.4 KiB db_name.db_user_data.tb_all_shards+0+8138464772.snappy.parquet
2022-06-03 03:26:20   35.3 KiB db_name.db_user_data.tb_all_shards+0+8138464852.snappy.parquet
2022-06-03 03:26:20   47.7 KiB db_name.db_user_data.tb_all_shards+0+8138464856.snappy.parquet
2022-06-03 03:26:20   35.1 KiB db_name.db_user_data.tb_all_shards+0+8138464926.snappy.parquet
2022-06-03 03:26:20   36.3 KiB db_name.db_user_data.tb_all_shards+0+8138464931.snappy.parquet
2022-06-03 03:26:21   33.5 KiB db_name.db_user_data.tb_all_shards+0+8138464949.snappy.parquet
2022-06-03 03:26:21   39.4 KiB db_name.db_user_data.tb_all_shards+0+8138464951.snappy.parquet
2022-06-03 03:26:21   36.3 KiB db_name.db_user_data.tb_all_shards+0+8138464977.snappy.parquet
2022-06-03 03:26:21   39.0 KiB db_name.db_user_data.tb_all_shards+0+8138464983.snappy.parquet
2022-06-03 03:26:22   33.4 KiB db_name.db_user_data.tb_all_shards+0+8138465007.snappy.parquet
2022-06-03 03:26:22   37.0 KiB db_name.db_user_data.tb_all_shards+0+8138465009.snappy.parquet
2022-06-03 03:26:22   35.8 KiB db_name.db_user_data.tb_all_shards+0+8138465025.snappy.parquet
2022-06-03 03:26:22   29.8 KiB db_name.db_user_data.tb_all_shards+0+8138465035.snappy.parquet

**Comment by sudeshwasnik** [comment]
@harveyyue QQ - `schema.compatibility` by default is NONE. The connector keeps writing to the same file only if two schemas are exactly the same, else a new file will be created. 
using `schema.compatibility` (BACKWARD or FORWARD) appropriately should only create new file when schema version change is incompatible. what value of `schema.compatibility` are you using ?

**Comment by sudeshwasnik** [comment]
That is, you have records in topic multiple schemas, all incompatible to each other. The ask is to create one file per schema, and not new file for every new schema encountered. 
The connector exposes `schema.compatibility` config, and the issue is explicitly when it is NONE, and connector should create one file per schema version (in a single rotation).

**Comment by harveyyue** [comment]
Yes, currently, the connector will create one new file when met a new schema version not stored in partition path map.
In [kafka-connect-storage-common](https://github.com/confluentinc/kafka-connect-storage-common) module, I add a property `rotate.multiple.schema` to decide whether create a new file, the default value `true` means keep the origin action for compatibility.

**Comment by dberardo-com** [comment]
is it possible to have a way to simply consider all fields as optional and have therefore parquet files where columns are always increased and values defaulted to null ?

i am talking about highly unstructured json payloads like:

`
{a: 1, b: 2}
{b: 4, c: 5}
{d: 5, :t: "hello"}
`

is it possible to save these 3 records in the same parquest / csv file on S3 ?
